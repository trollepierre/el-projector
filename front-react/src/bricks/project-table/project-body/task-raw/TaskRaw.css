.task-raw {
    display: grid;
    justify-content: space-around;
    grid-template-columns: repeat(10, 1fr [col-start]);
    align-items: center;
    justify-items: center;
    height: 50px;
}

td {
    justify-content: center;
    justify-items: center;
    display: flex;
}
