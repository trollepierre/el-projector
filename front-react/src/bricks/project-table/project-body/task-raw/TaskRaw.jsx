import React from 'react';
import Button from '../../../../components/button/Button';
// eslint-disable-next-line no-unused-vars
import { eliminate, differ, remove } from '../../services/task-edit';
// eslint-disable-next-line no-unused-vars
import styles from './TaskRaw.css'

 const TaskRaw = ({ task }) => (
   <tr className="task-raw">
     <td className="id">
       {task.id}
     </td>
     <td className="name">
       {task.name}
     </td>
     <td className="description">
       {task.description}
     </td>
     <td className="points">
       {task.points}
     </td>
     <td className="optimal-date">
       {task.optimalDate}
     </td>
     <td className="priority">
       {task.priority}
     </td>
     <td className="benefit">
       {task.benefit}
     </td>
     <td className="needed-time">
       {task.neededTime}
     </td>
     {/*<td className="epic-id">*/}
     {/*  {task.epicId}*/}
     {/*</td>*/}
     {/*<td className="needed-material">*/}
     {/*  {task.neededMaterial}*/}
     {/*</td>*/}
     {/*<td className="sub-task-ids">*/}
     {/*  {task.subTaskIds.map(id => (<p key={id}>{id}</p>))}*/}
     {/*</td>*/}
     {/*<td className="master-task-id">*/}
     {/*  {task.masterTaskId}*/}
     {/*</td>*/}
     <td>
       <label htmlFor='done'>Done</label>
       <input type='checkbox' id='done'></input>
     </td>
     <td className="impact">
       <Button
         background-color="orange"
         color="white"
         text="later"
         onClick={differ}
       />
     </td>
     {/*<td>*/}
     {/*  <Button*/}
     {/*    background-color="red"*/}
     {/*    color="white"*/}
     {/*    text="not doing"*/}
     {/*    onClick={eliminate}*/}
     {/*  />*/}
     {/*</td>*/}
     {/*<td>*/}
     {/*  <Button*/}
     {/*    background-color="red"*/}
     {/*    color="white"*/}
     {/*    text="delete"*/}
     {/*    onClick={remove}*/}
     {/*  />*/}
     {/*</td>*/}
   </tr>
 );

export default TaskRaw;
