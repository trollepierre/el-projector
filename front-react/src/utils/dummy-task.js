const dummyTask = (task) => ({
  id: 1,
  name: 'Build el-projector',
  description: 'more data about how to do it',
  points: 999, // given by server
  createdDate: '19/05/2019',
  endDate: undefined, // must be done before this date
  optimalDate: undefined, // user would do it at this date
  isDone: false,
  doneDate: undefined, // for log record
  priority: 'HIGHLY_IMPORTANT', // Enum with 5 levels (inherited by default, but can be updated)
  benefit: 'ORGASM_HAPPINESS', // Enum with 5 levels
  neededTime: 'ONE_MONTH', // Enum
  epicId: 'A1', // Yearly happiness epic
  neededMaterial: ['MacTo', 'iMoi', 'MacNo', 'home'],
  subTaskIds: ['TASK10', 'TASK11', 'TASK12'], // checklist of subtasks that depends on it
  masterTaskId: 'TASK0', // this task depends on the master one
  ...task,
})

export { dummyTask }
