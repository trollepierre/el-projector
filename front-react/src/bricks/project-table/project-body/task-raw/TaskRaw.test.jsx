import React from 'react';
import { shallow } from 'enzyme'
import TaskRaw from './TaskRaw';
import { dummyTask } from '../../../../utils/dummy-task';

describe('TaskRaw', () => {
  describe('template', () => {
    it('should match snapshot', () => {
      // Given
      const props = {
        task: dummyTask()
      }

      // When
      const wrapper = shallow(<TaskRaw {...props}/>)

      // Then
      expect(wrapper).toMatchSnapshot()
    });
  });
});
