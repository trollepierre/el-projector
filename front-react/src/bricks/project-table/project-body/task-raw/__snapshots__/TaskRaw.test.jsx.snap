// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskRaw template should match snapshot 1`] = `
<tr
  className="task-raw"
>
  <td
    className="id"
  >
    1
  </td>
  <td
    className="name"
  >
    Build el-projector
  </td>
  <td
    className="description"
  >
    more data about how to do it
  </td>
  <td
    className="points"
  >
    999
  </td>
  <td
    className="optimal-date"
  />
  <td
    className="priority"
  >
    HIGHLY_IMPORTANT
  </td>
  <td
    className="benefit"
  >
    ORGASM_HAPPINESS
  </td>
  <td
    className="needed-time"
  >
    ONE_MONTH
  </td>
  <td>
    <label
      htmlFor="done"
    >
      Done
    </label>
    <input
      id="done"
      type="checkbox"
    />
  </td>
  <td
    className="impact"
  >
    <Button
      background-color="orange"
      color="white"
      onClick={[Function]}
      text="later"
    />
  </td>
</tr>
`;
